Ports TGStation's Janitor Tools

nearlyNonexistent opened the pull request:
🆑nearlyNon
rscadd: Adds the holosign creator, advanced mop, trashbag of holding; all are able to be made in research labs.
rscadd: Janibelt from TGstation, spawns in the lockers.
/🆑

CakeQ commented:
please ensure the janibelt matches the other belt sprites

nearlyNonexistent commented:
only if you give me sprite to do so

FreePrivateer commented:
Or better yet, ditch the Janibelt, and add more things that can go on the equipment belt. Then we wouldn't require a new sprite. Well.... other than the inevitable equipment sprites.

Anticept commented:
One more thing that would be wonderful, as part of this, is TG style light replacers. When used, they would replace the light and consume the broken one, instead of dropping it on the floor.

Not sure if janitors on bay get replacers.

FreePrivateer commented:
@Anticept There are light replacers in each of the custodial lockers; They replace lights, are refilled through glass sheets, and leave the broken light on the ground.

nearlyNonexistent commented:
Main thing is the sprite requirement.

Should I remove janibelt from this PR and leave the advanced tools? Because you know if I do that it basically means nothing will end up getting done on the janibelt.

CakeQ commented:
Making an item sprite for the janibelt is as easy as recoloring one of the already existing belt sprites.
Literally 0 effort.

nearlyNonexistent commented:
equipment sprites for in belt tho

On Tue, Sep 4, 2018, 09:22 Daniel Thompson <notifications@github.com> wrote:

> Making an item sprite for the janibelt is as easy as recoloring one of the
> already existing belt sprites.
> Literally 0 effort.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Baystation12/Baystation12/pull/22945#issuecomment-418385781>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AD2rXfY5xA3t1tWtR8U4dvifz1_nQEx0ks5uXoy0gaJpZM4WYKkc>
> .
>

CakeQ commented:
@nearlyNonexistent not required. Nice, but not required.

nearlyNonexistent commented:
made it a regular belt just recolored purple,pushing commit.

nearlyNonexistent commented:
something broke when I changed the sprite hold on

nearlyNonexistent commented:
nope, it's working, Travis is just shitting itself before even trying to compile it.

Anticept commented:
@FreePrivateer yes i know, but it's the part where they drop the lights on the ground instead of just consuming them for the equivalent glass. It would be a quality of life thing if the replacers would just do that.

babydoll commented:
Merge conflicts. The trashbag thing is pretty wacky, personally, but I guess it probably won't ever be used anyway

nearlyNonexistent commented:
it takes diamonds so probably not. why make one of those when you can make
a bag of holding

On Fri, Sep 7, 2018, 01:09 Sabira <notifications@github.com> wrote:

> Merge conflicts. The trashbag thing is pretty wacky, personally, but I
> guess it probably won't ever be used anyway
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Baystation12/Baystation12/pull/22945#issuecomment-419333454>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AD2rXa7mCuW3MHkqSjweYt7tpcNxEcA9ks5uYg2rgaJpZM4WYKkc>
> .
>

Anticept commented:
Because a trashbag can grab a whole pile of junk at once just by clicking the tile.

A bag of holding cannot.

Hardly a justified use case but there it is.

babydoll reviewed (dismissed):
You deleted the plunger in-hand icons when you resolved the conflict

nearlyNonexistent commented:
shit

PsiOmegaDelta commented on code/game/objects/items/weapons/mop.dm:
```diff
@@ -42,3 +43,41 @@
+	var/refill_rate = 1 //Rate per process() tick mop refills itself
+	var/refill_reagent = /datum/reagent/water //Determins what reagent to use for refilling, just in case someone wanted to make a HOLY MOP OF PURGING
+
+/obj/item/weapon/mop/advanced/New()
```
`/Initialize()`

PsiOmegaDelta commented on code/game/objects/items/weapons/mop.dm:
```diff
@@ -42,3 +43,41 @@
+	var/refill_reagent = /datum/reagent/water //Determins what reagent to use for refilling, just in case someone wanted to make a HOLY MOP OF PURGING
+
+/obj/item/weapon/mop/advanced/New()
+	..()
```
`. = ..()`

PsiOmegaDelta commented on code/game/objects/structures/holosigns.dm:
```diff
@@ -0,0 +1,35 @@
+	var/obj/item/holosign_creator/projector
+	icon_state = "holosign"
+
+/obj/structure/holosign/New(loc, source_projector)
```
`/obj/structure/holosign/Initialize(var/maploading, var/source_projector)`

PsiOmegaDelta commented on code/game/objects/structures/holosigns.dm:
```diff
@@ -0,0 +1,35 @@
+	if(source_projector)
+		projector = source_projector
+		projector.signs += src
+	..()
```
`. = ..()`

PsiOmegaDelta commented on code/game/objects/items/holosign_creator.dm:
```diff
@@ -0,0 +1,54 @@
+	throwforce = 0
+	throw_speed = 3
+	throw_range = 7
+	var/list/signs = list()
```
This list has to be nulled during `/Destroy()` or it'll keep references to signs unnecessarily.

PsiOmegaDelta commented on code/game/objects/structures/holosigns.dm:
```diff
@@ -0,0 +1,35 @@
+	if(.)
+		return
+	visible_message(SPAN_NOTICE("\The [user] waves through \the [src], causing it to dissipate."))
+	deactivate(user)
```
Why not call `attackby()` instead of copy-pasting?

nearlyNonexistent replied:
for the different message?

nearlyNonexistent commented:
I can't find the indentation error :(

nearlyNonexistent committed:
Janitor Tools Merged Commit

🆑nearlyNon
rscadd: Adds the holosign creator, advanced mop, trashbag of holding; all are able to be made in research labs.
rscadd: Janibelt from TGstation, spawns in the lockers.
/🆑

nearlyNonexistent commented:
ready for review now merge conflicts gone and it passes travis now
